src/index.ts

```diff
@@ -15,6 +15,10 @@ export { default as Modal } from './Modal'
 import { ModalProps as TModalProps } from './Modal'
 export type ModalProps = TModalProps
 
+// Actions > ModalLegacy
+import { ModalLegacyProps } from './Modal'
+export type ModalLegacyProps
+
```

Comment:
The name `ModalLegacyProps` will conflict with declaration below it when you export the type. Try this instead:
```suggestion
import { ModalLegacyProps as TModalLegacyProps } from './Modal'
export type ModalLegacyProps = TModalLegacyProps
```